Fix precommit check failures with missingJavadoc task

**Describe the bug**
The `missingJavadoc` task in [javadoc branch](https://github.com/opensearch-project/OpenSearch/tree/javadoc) currently works well when run on its own (i.e. `./gradlew missingJavadoc`) but fails the precommit checks due to some wrangled gradle build logic. This can be reproduced by running `./gradlew precommit` on `javadoc` branch. The `validateNebulaPom` task, which is part of precommit checks, fails because it looks for pom file in the wrong location (`build/publications/nebula/pom-default.xml`) when the actual pom file is generated in the path configured [here](https://github.com/opensearch-project/OpenSearch/blob/javadoc/buildSrc/src/main/java/org/opensearch/gradle/PublishPlugin.java#L76) (e.g. for benchmarks module`build/distributions/opensearch-benchmarks-7.10.3-SNAPSHOT.pom`). On removing this [line](https://github.com/opensearch-project/OpenSearch/blob/javadoc/build.gradle#L51), the precommit logic seems to work fine. Somehow, applying missingJavadoc script plugin here is causing the PomValidation task to read pom from default location instead of overridden location.

**To Reproduce**
Steps to reproduce the behavior:
1. Sync remote `javadoc` branch
2. Run `./gradlew precommit`. It should fail in PomValidation
3. Comment this line.
4. Run step 2 again. It should pass now.

**Expected behavior**
`precommit` task should work fine with `missingJavadoc` gradle task.

**Host/Environment (please complete the following information):**
 - OS: iOS
 - Version: 10.14.6

**Logs**
```
➜  search git:(javadoc) ./gradlew precommit

> Configure project :
========================= WARNING =========================
         Backwards compatibility tests are disabled!
See https://github.com/opensearch-project/OpenSearch/issues/105
===========================================================

> Configure project :qa:os
Cannot add task 'destructiveDistroTest.docker' as a task with that name already exists.
=======================================
OpenSearch Build Hamster says Hello!
  Gradle Version        : 6.6.1
  OS Info               : Mac OS X 10.14.6 (x86_64)
  JDK Version           : 14 (OpenJDK)
  JAVA_HOME             : /Library/Java/JavaVirtualMachines/jdk-14.jdk/Contents/Home
  Random Testing Seed   : C2BC3255C3EA0731
  In FIPS 140 mode      : false
=======================================

> Task :benchmarks:validateNebulaPom FAILED

FAILURE: Build failed with an exception.

* What went wrong:
A problem was found with the configuration of task ':benchmarks:validateNebulaPom' (type 'PomValidationTask').
> File '/Users/setiah/projects/odfe/search/benchmarks/build/publications/nebula/pom-default.xml' specified for property 'pomFile' does not exist.
```

--- a/client/rest/src/main/java/org/opensearch/client/NodeSelector.java
+++ b/client/rest/src/main/java/org/opensearch/client/NodeSelector.java
@@ -54,7 +54,7 @@ public interface NodeSelector {
      * <p>
      * Implementers should not rely on the ordering of the nodes.
      *
-     * @param nodes the {@Node}s targeted for the sending requests
+     * @param nodes the {@link Node}s targeted for the sending requests
      */
     void select(Iterable<Node> nodes);
     /*

--- a/client/rest/src/main/java/org/opensearch/client/RestClientBuilder.java
+++ b/client/rest/src/main/java/org/opensearch/client/RestClientBuilder.java
@@ -179,7 +179,7 @@ public final class RestClientBuilder {
      * @param pathPrefix the path prefix to be cleaned up.
      * @return the cleaned up path prefix.
      * @throws NullPointerException if {@code pathPrefix} is {@code null}.
-     * @throws IllegalArgumentException if {@pathPrefix} is empty, or ends with more than one '/'.
+     * @throws IllegalArgumentException if {@code pathPrefix} is empty, or ends with more than one '/'.
      */
     public static String cleanPathPrefix(String pathPrefix) {
         Objects.requireNonNull(pathPrefix, "pathPrefix must not be null");

--- a/gradle/missing-javadoc.gradle
+++ b/gradle/missing-javadoc.gradle
@@ -52,20 +52,22 @@ allprojects {
       docroot = file("${buildDir}/site")
     }
 
-    // TODO: Enable missingJavadoc task
+    // TODO: Add missingJavadoc checks to precommit plugin
     // See https://github.com/opendistro-for-elasticsearch/search/issues/221
-    // Once we get the javadocs ready for the code base
-    // we will disable the default javadoc task and use our own
-    // missingJavadoc task below. The default javadoc task
-    // will just invoke 'missingJavadoc' (to allow people to call
-    // conventional task name).
-    tasks.matching { it.name == "javadoc" }.all {
+    // Currently the missingJavadoc task fails due to missing documentation
+    // across multiple modules. Once javadocs are added, we can
+    // add this task to precommit plugin.
+    tasks.withType(MissingJavadocTask).configureEach {
       enabled = true
-      // enabled = false
-      // dependsOn "missingJavadoc"
+      title = "${project.rootProject.name} ${project.name} API"
+
+      // Set up custom doclet.
+      dependsOn configurations.missingdoclet
+      docletpath = configurations.missingdoclet
     }
 
-    task missingJavadoc(type: MissingJavadocTask) {
+
+    tasks.register('missingJavadoc', MissingJavadocTask) {
       description "This task validates and generates Javadoc API documentation for the main source code."
       group "documentation"
 
@@ -76,16 +78,6 @@ allprojects {
 
       outputDir = project.javadoc.destinationDir
     }
-  }
-}
-
-allprojects {
-  project.tasks.withType(MissingJavadocTask) {
-    title = "${project.rootProject.name} ${project.name} API"
-
-    // Set up custom doclet.
-    dependsOn configurations.missingdoclet
-    docletpath = configurations.missingdoclet
   }
 }
 

